### PR TITLE
More operands 2

### PR DESCRIPTION
**Pull request by lima-limon-inc**
- Added LDI
- Added LD
- Added AND
- Added NOT

### CODE REVIEW

File: src/virtual_machine.rs [hunk 1]
```diff
@@ -225,30 +306,71 @@ impl VM {
...
+            Opcode::OpAdd {
                 dr,
                 sr1,
                 second_arg,
             } => {
                 let sr1_val = self.value_from_register(sr1);
                 let second_value = match second_arg {
-                    AddMode::IMMEDIATE { imm5 } => {
+                    Mode::IMMEDIATE { imm5 } => {
                         let value = sign_extend(imm5, 5);
```
Comment (damiramirez) [inline src/virtual_machine.rs]: You´re calling the `sign_extend` function in decode and execute. It's not necessary to call it here.

File: src/virtual_machine.rs [hunk 2]
```diff
@@ -225,30 +306,71 @@ impl VM {
...
+            }
+            Opcode::OpAnd {
+                dr,
+                sr1,
+                second_arg,
+            } => {
+                let sr1_val = self.value_from_register(sr1);
+                let second_value = match second_arg {
+                    Mode::IMMEDIATE { imm5 } => {
+                        let value = sign_extend(imm5, 5);
```
Comment (damiramirez) [inline src/virtual_machine.rs]: Same here

**Review by damiramirez — APPROVED** [review]
LGTM  + The comments that I have added do not impact but it would be a good idea to remove it.

File: src/virtual_machine.rs (hunk 1 above)
Comment (lima-limon-inc) [reply src/virtual_machine.rs]: Thanks Dami! Great catch

**Pull request merged**